edge-modules/TestAnalyzer/Program.cs

```diff
@@ -32,23 +34,42 @@ static async Task Main(string[] args)
             await CreateWebHostBuilder(args).Build().RunAsync(cts.Token);
         }
 
+        static async Task<DateTime> LoadStartupTimeFromStorage()
+        {
+            DateTime lastReceivedAt = DateTime.MinValue;
+            await ReportingCacheWithStorage.Instance.Init(Settings.Current.StoragePath, Settings.Current.OptimizeForPerformance);
+            var messagesSnapShot = ReportingCache.Instance.GetMessagesSnapshot();
+            foreach (KeyValuePair<string, IList<SortedSet<MessageDetails>>> moduleMesssages in messagesSnapShot)
+            {
+                foreach (var moduleMesssage in moduleMesssages.Value)
+                {
+                    if (lastReceivedAt < moduleMesssage.Last().EnqueuedDateTime)
+                    {
+                        lastReceivedAt = moduleMesssage.Last().EnqueuedDateTime;
+                    }
+                }
+            }
+
+            return lastReceivedAt;
+        }
+
         static IWebHostBuilder CreateWebHostBuilder(string[] args) =>
             WebHost.CreateDefaultBuilder(args)
                 .UseUrls($"http://*:{Settings.Current.WebhostPort}")
                 .UseStartup<Startup>();
 
-        static async Task ReceiveMessages()
+        static async Task ReceiveMessages(DateTime lastReceivedMesssage)
```

Comment:
```suggestion
        static async Task ReceiveMessagesAsync(DateTime lastReceivedMesssage)
```

Reply:
I will search for async and follow this pattern elsewhere